zerver/views/video_calls.py

```diff
@@ -151,3 +153,35 @@ def deauthorize_zoom_user(request: HttpRequest) -> HttpResponse:
             auth=(settings.VIDEO_ZOOM_CLIENT_ID, settings.VIDEO_ZOOM_CLIENT_SECRET),
         ).raise_for_status()
     return json_success()
+
+
+@zulip_login_required
+@never_cache
+def get_bbb_url(request: HttpRequest, user_profile: UserProfile) -> HttpResponse:
+    id = "zulip-" + str(random.randint(100000000000, 999999999999))
+    password = ''.join(random.SystemRandom().choice(string.ascii_uppercase + string.digits) for _ in range(10))
+    checksum = hashlib.sha1(("create" + "meetingID=" + id + "&moderatorPW=" + password + settings.BIG_BLUE_BUTTON_SECRET).encode()).hexdigest()
+    return json_success({"url": "/calls/bbb/join?meeting_id=%22" + id + "%22&password=%22" + password + "%22&checksum=%22" + checksum + "%22"})
+
+
+@zulip_login_required
```

Comment:
Do we want `zulip_login_required` here?  It would seem better to not do so, so that users can open the meeting in a different browser.  (I.e. just having access to the meeting URL containing ID/password would be sufficient).

Reply:
As this methods sets the name of the user in Big Blue Button I do think that is does need a user to be logged in

Reply:
Ahh, OK, makes sense; so we're doing this not for security (we're notably not checking the user is logged in to the organization where the link was created, for example), just to get a bit of data.